Analysis/Tasks/PWGCF/FemtoDream/include/FemtoDream/FemtoDreamV0Selection.h

```diff
@@ -0,0 +1,248 @@
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file FemtoDreamV0Selection.h
+/// \brief Definition of the FemtoDreamV0Selection
+/// \author Valentina Mantovani Sarti, TU München valentina.mantovani-sarti@tum.de and Andi Mathis, TU München, andreas.mathis@ph.tum.de
+
+#ifndef ANALYSIS_TASKS_PWGCF_FEMTODREAM_FEMTODREAMV0SELECTION_H_
+#define ANALYSIS_TASKS_PWGCF_FEMTODREAM_FEMTODREAMV0SELECTION_H_
+
+#include "FemtoDreamObjectSelection.h"
+#include "FemtoDreamTrackSelection.h"
+#include "FemtoDreamSelection.h"
+
+#include "ReconstructionDataFormats/PID.h"
+#include "AnalysisCore/RecoDecay.h"
+#include "Framework/HistogramRegistry.h"
+#include <Rtypes.h>
+#include <cmath>
+
+using namespace o2::framework;
+
+namespace o2::analysis
+{
+namespace femtoDream
+{
+namespace femtoDreamV0Selection
+{
+enum V0Sel { kpTV0Min,
+             kpTV0Max,
+             kDCAV0DaughMax,
+             kCPAV0Min,
+             kTranRadV0Min,
+             kTranRadV0Max,
+             kDecVtxMax };
+enum ChildTrackType { kPosTrack,
+                      kNegTrack };
+} // namespace femtoDreamV0Selection
+
+/// \class FemtoDreamV0Selection
+/// \brief Cut class to contain and execute all cuts applied to V0s
+class FemtoDreamV0Selection : public FemtoDreamObjectSelection<float, femtoDreamV0Selection::V0Sel>
+{
+ public:
+  /// Initializes histograms for the task
+  void init(HistogramRegistry* registry);
+
+  template <typename T>
+  void setPIDSpecies(T& pids) ///K0s,Λ and antiΛ
+  {
+    std::vector<int> tmpPids = pids; // necessary due to some features of the configurable
+    for (const o2::track::PID& pid : tmpPids) {
+      mPIDspecies.push_back(pid);
+    }
+  }
+
+  template <typename C, typename V, typename T>
+  bool isSelectedMinimal(C const& col, V const& v0, T const& posTrack, T const& negTrack);
+
+  template <typename C, typename V, typename T>
+  std::vector<uint64_t> getCutContainer(C const& col, V const& v0, T const& posTrack, T const& negTrack);
+
+  template <typename C, typename V>
+  void fillQA(C const& col, V const& v0);
+
+  template <typename T1, typename T2>
+  void setChildCuts(femtoDreamV0Selection::ChildTrackType child, T1 selVal, T2 selVar, femtoDreamSelection::SelectionType selType)
+  {
+    if (child == femtoDreamV0Selection::kPosTrack)
+      PosDaughTrack.setSelection(selVal, selVar, selType);
+    else if (child == femtoDreamV0Selection::kNegTrack)
+      NegDaughTrack.setSelection(selVal, selVar, selType);
+  }
+
+ private:
+  FemtoDreamTrackSelection PosDaughTrack;
+  FemtoDreamTrackSelection NegDaughTrack;
+  std::vector<o2::track::PID> mPIDspecies;
```

Comment:
Temove

Reply:
Here I remove only the mPIDspecies since it is not used at all.
The FemtoDreamTrackSelection are used in the class.

Reply:
perfect, thanks!